raiden/transfer/node.py

```diff
@@ -475,24 +499,16 @@ def handle_token_network_action(
 
 
 def handle_delivered(chain_state: ChainState, state_change: ReceiveDelivered) -> TransitionResult:
-    # TODO: improve the complexity of this algorithm
-    queueids_to_remove = []
-    for queueid, queue in chain_state.queueids_to_queues.items():
-        if queueid.channel_identifier == CHANNEL_IDENTIFIER_GLOBAL_QUEUE:
-            filtered_queue = [
-                message
-                for message in queue
-                if message.message_identifier != state_change.message_identifier
-            ]
-
-            if not filtered_queue:
-                queueids_to_remove.append(queueid)
-            else:
-                chain_state.queueids_to_queues[queueid] = filtered_queue
-
-    for queueid in queueids_to_remove:
-        del chain_state.queueids_to_queues[queueid]
-
+    """ Check if the "Delivered" message exists in the global queue and delete if found."""
+    # Find the QueueIdentifier with channel_identifier == CHANNEL_IDENTIFIER_GLOBAL_QUEUE
+    queueid = list(filter(
+        lambda i: i.channel_identifier == CHANNEL_IDENTIFIER_GLOBAL_QUEUE,
```

Comment:
There's one global queue per receiver, so you don't need to iterate, and instead, do something like:
```
global_queue_id = QueueIdentifier(receiver=state_change.sender, channel_id=CHANNEL_IDENTIFIER_GLOBAL_QUEUE)
if global_queue_id in chain_state.queueids_to_queues:  # you need to check if the queue exists
    inplace_delete_message...
```

Reply:
Also notice that not all messages are cleaned on `Delivered`, only those in this global-per-receiver queue. If there's no corresponding message in that queue, that's ok, it may be on the specific queue and the same goes for `handle_processed`